pkg/operator/ceph/object/user.go

```diff
@@ -54,8 +58,16 @@ func decodeUser(data string) (*ObjectUser, int, error) {
 		return nil, RGWErrorParse, errors.Wrapf(err, "failed to unmarshal json. %s", data)
 	}
 
-	rookUser := ObjectUser{UserID: user.ID, DisplayName: &user.DisplayName, Email: &user.Email}
+	rookUser := ObjectUser{UserID: user.ID, DisplayName: &user.DisplayName, Email: &user.Email, MaxBuckets: *user.MaxBuckets}
 
+	if len(user.Caps) > 0 {
+		rookUser.Caps = user.Caps
+	}
+
+	if (user.UserQuota != admin.QuotaSpec{}) && (*user.UserQuota.Enabled) {
```

Comment:
I think we need something like this to avoid nil refs:
```suggestion
	if user.UserQuota.Enabled != nil && *user.UserQuota.Enabled {
		if user.UserQuota.MaxQuota != nil {
			rookUser.MaxSize = int(*user.UserQuota.MaxSize)
		}
		if user.UserQuota.MaxObjects != nil {
			rookUser.MaxObjects = int(*user.UserQuota.MaxObjects)
		}
	}
```